waspc/src/Wasp/AppSpec/Core/IsDecl.hs

```diff
@@ -0,0 +1,16 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+
+module Wasp.AppSpec.Core.IsDecl
+  ( IsDecl (..),
+  )
+where
+
+import Data.Typeable (Proxy (Proxy), Typeable, typeRep)
+
+class (Typeable a) => IsDecl a where
```

Comment:
I was shifting things around a couple of times, so I can't remember exactly, but I believe it was IsDecl ---importedby--> Ref ---importedby--> AppSpec types (Api, Route, ...) ---importedby--> Decl . And before IsDecl was part of Decl, so that would be circ dep, since it starts and ends with Decl.